### PR TITLE
Update 6-1 to follow BASHER

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* Update README
* Update all dependencies and framework versions
* Support CAE events

## Does this introduce a breaking change?

```
[ x ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ x ] Code style update (formatting, local variables)
[ x ] Refactoring (no functional changes, no api changes)
[ x ] Documentation content changes
[ x ] Other... Please describe: dependency updates
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -1,94 +1,116 @@
...
         },
         {
-          "key": "TenantId",
-          "value": "$tenantId"
+            "Id": "client",
+            "Name": "ProfileSPA",
+            "Kind": "SinglePageApplication",
+            "Audience": "AzureADMyOrg",
+            "HomePage": "https://localhost:4200",
+            "ReplyUrls": "http://localhost:4200, https://localhost:4200/auth",
```
Comment (derisen) [inline 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json:59]:
```suggestion
            "ReplyUrls": "http://localhost:4200, http://localhost:4200/auth",
```

Comment (v-michaelmi) [inline 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json:59]: `https://localhost:4200/auth` doesn't match with `http://localhost:4200/auth` in the code.

File: 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1 [hunk 2]
```diff
@@ -205,231 +199,342 @@ Function CreateAppRole([string] $types, [string] $name, [string] $description)
...
 
-   # URL of the AAD application in the Azure portal
-   # Future? $clientPortalUrl = "https://portal.azure.com/#@"+$tenantName+"/blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
-   $clientPortalUrl = "https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/CallAnAPI/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
-   Add-Content -Value "<tr><td>client</td><td>$currentAppId</td><td><a href='$clientPortalUrl'>ProfileSPA</a></td></tr>" -Path createdApps.html
+    # Configure known client applications for service 
+    Write-Host "Configure known client applications for the 'service'"
+    $knowApplications = New-Object System.Collections.Generic.List[System.String]
+    $knowApplications.Add($clientAadApplication.AppId)
+    Update-MgApplication -ApplicationId $currentAppObjectId -Api @{KnownClientApplications = $knowApplications}
```
Comment (v-michaelmi) [inline 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1:445]: Written as a reminder that we need to update the `$currentAppObjectId` to be pointed to the API project. Right now this is sending the SPA as a known client to the SPA application in Azure.

**Comment by v-michaelmi** [comment]
README has the following line:

`cd 7-AdvancedScenarios/1-call-api-obo/API/ProfileAPI`

Should be `cd 6-AdvancedScenarios/1-call-api-obo/API/ProfileAPI`

File: 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json (hunk 1 above)
Comment (derisen) [reply 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json:59]: done!

File: 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1 (hunk 2 above)
Comment (derisen) [reply 6-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1]: done!

File: 6-AdvancedScenarios/1-call-api-obo/README.md [hunk 3]
```diff
@@ -35,51 +58,55 @@ This sample demonstrates an Angular single-page application which lets a user au
...
 > :warning: To avoid path length limitations on Windows, we recommend cloning into a directory near the root of your drive.
 
 ### Step 2. Install .NET Core API dependencies
 
 ```console
-    cd ms-identity-javascript-angular-tutorial
-    cd 7-AdvancedScenarios/1-call-api-obo/API/ProfileAPI
-    dotnet restore
+   cd ms-identity-javascript-angular-tutorial
+   cd 7-AdvancedScenarios/1-call-api-obo/API/ProfileAPI
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/1-call-api-obo/README.md:89]: >7-AdvancedScenarios/1-call-api-obo/API/ProfileAPI [](http://example.com/codeflow?start=6&length=49)

cd 6-AdvancedScenarios/1-call-api-obo/API/ProfileAPI

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: 6-AdvancedScenarios/1-call-api-obo/README.md (hunk 3 above)
Comment (derisen) [reply 6-AdvancedScenarios/1-call-api-obo/README.md:89]: done

**Pull request merged**